funcx_endpoint/funcx_endpoint/endpoint/interchange.py

```diff
@@ -239,131 +197,53 @@ def start_executors(self):
             if hasattr(executor, "passthrough") and executor.passthrough is True:
                 executor.start(results_passthrough=self.results_passthrough)
 
-    def apply_reg_info(self, reg_info):
-        self.client_address = reg_info["public_ip"]
-        self.client_ports = (
-            reg_info["tasks_port"],
-            reg_info["results_port"],
-            reg_info["commands_port"],
-        )
-
     def register_endpoint(self):
         reg_info = register_endpoint(
             self.funcx_client, self.endpoint_id, self.endpoint_dir, self.endpoint_name
         )
-        self.apply_reg_info(reg_info)
-        return reg_info
+        self.connection_params = reg_info
 
-    def migrate_tasks_to_internal(self, quiesce_event):
+    def migrate_tasks_to_internal(
+        self,
+        connection_params: pika.connection.Parameters,
+        endpoint_uuid: str,
+        pending_task_queue: multiprocessing.Queue,
+        quiesce_event: multiprocessing.Event,
```

Comment:
At the usage site, I see this being called as
```python
self.migrate_tasks_to_internal(
    self.connection_params,
    self.endpoint_id,
    self.pending_task_queue,
    self._quiesce_event
)
```

So why not remove all of the arguments from this method and make it `self.migrate_tasks_to_internal()`? Is there a reason that it must be possible to pass, for example, distinct pika parameters to this method?